vm/src/format.rs

```diff
@@ -281,14 +282,22 @@ impl FormatSpec {
         separator: char,
     ) -> String {
         let mut result = String::new();
-        let mut remaining: usize = magnitude_string.len();
-        for c in magnitude_string.chars() {
+
+        // Don't add separators to the floating decimal point of numbers
```

Comment:
I'm not sure exactly what the functionality of this is supposed to be, but you might want to do `.splitn('.', 2)` so that the splitting sort of has right associativity. Then you don't have to `.collect()` it and you can do `let magnitude_integer_string = parts.next().unwrap();` and `if let Some(part) = parts.next() { ... }`

Reply:
This is needed to prevent adding an incorrect separator to floats, as this is what it looked like before:
```Python
>>>>> '{:,.2f}'.format(1234567890.1234)
'1,234,567,890,.12'
```
With this change, it's only adding the separator to the integer portion of the number:
```Python
>>>>> '{:,.2f}'.format(1234567890.1234)
'1,234,567,890.12'
```

I'll change it to the way you suggested, as that is cleaner.